### PR TITLE
Added event to ColabToken.sol

### PR DESCRIPTION
**Pull request by Colin-Stark**
Added events to help users know when transactions like mint and transfer was done

### CODE REVIEW

**Comment by Colin-Stark** [comment]
😄 Hello **[pelzfx](https://github.com/pelzfx)**

I have viewed your project. i made some adjustments

# Fix and Improvement
1. I added a maximum number of tokens that can be minted which was used to avoid unlimited minting and also minting beyond supply.
2. The `tokenURI` function can throw an error if called with a token ID that does not exist.
3. The contract does not define a fallback function. This can lead to accidental transfers of tokens to the contract, which would then be irretrievable
4. Previously i had also added events to ensure transparency.


# Suggestion
The following suggestions were made along with codes to show how it should be done 
 - I added a fee to the safeMint function to incentivize users to hold the tokens and prevent spamming the minting function.
 - I added a feature to enable token holders transfer their token to other user.

Other  suggestions which were not coded are

 - There should be an implementation for token holders to lock their token which would be for staking purpose and give them governance right 

# CONCLUSION 
In conclusion i love to say this is an amazing project with prospects and i will like to see it grow further

**Comment by smcgive** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
ScribbleReader, @zishansami102, @Nasafato, @bt3gl, @vanohj, @CJ42, @hunterm01, @shintemy, @JhnEngblm, @olivermontes, @palsp, @mvkvc, @NathanBSC, @eliobricenov, @nametsego, @philipglazman, @abhinavprasad47, @PeiQi0, @anthowen, @NOOMA-42
